htlcswitch/switch.go

```diff
@@ -2269,12 +2279,48 @@ func (s *Switch) HasActiveLink(chanID lnwire.ChannelID) bool {
 // returns after the link has been completely shutdown.
 func (s *Switch) RemoveLink(chanID lnwire.ChannelID) {
 	s.indexMtx.Lock()
-	link := s.removeLink(chanID)
+	link, _ := s.getLink(chanID)
```

Comment:
nit: can check `if link == nil` here to exit early then we won't need to check `link != nil` below.

Reply:
also log the error here so it's easier for debugging?

Reply:
`RemoveLink` is called on link startup always so if it's not found an error will always be returned. Maybe a trace log just so users don't have logs that look like errors?

Reply:
Yap, as long as we can debug it, anything would be fine.

> RemoveLink is called on link startup always so if it's not found an error will always be returned. 

I do see several other places calling `RemoveLink` tho.

Reply:
Right but most of the time it's going to log an error just because of a reconnect

Reply:
added trace log